[14.0][IMP] product_template_tags: usability improvements

ivantodorovich opened the pull request:
A batch of small improvements on this module


* Added sequence field
* Fixed company security rules
* Improved tags form view

before:
![form_before](https://user-images.githubusercontent.com/1914185/124946373-5c102980-dfe5-11eb-9e2c-2ff40992f440.png)

after:
![form_after](https://user-images.githubusercontent.com/1914185/124946390-5e728380-dfe5-11eb-9341-2d5865fa6bfd.png)



* Improved tags tree view
* Improved product.template kanban view

before:
![before](https://user-images.githubusercontent.com/1914185/124946456-6af6dc00-dfe5-11eb-8452-50da0b0fc3d6.png)

after:
![after](https://user-images.githubusercontent.com/1914185/124946466-6cc09f80-dfe5-11eb-9f3a-b8b263c07d24.png)


* Added tags product.product kanban view
* Removed menu item from general configuration
* Added glue modules to show Product Tags menues in specific apps

| _before: (in general settings)_ | _after: (in glue modules)_ |
| --- | --- |
| ![removed_menu](https://user-images.githubusercontent.com/1914185/124946633-9083e580-dfe5-11eb-80e5-8d8f1502cc8a.png) | ![image](https://user-images.githubusercontent.com/1914185/124946864-bf9a5700-dfe5-11eb-8f1b-b4f949f38155.png)
 |

legalsylvain commented on product_template_tags/models/product_template_tag.py:
```diff
@@ -5,11 +5,11 @@
     _description = "Product Tag"
 
     name = fields.Char(string="Name", required=True, translate=True)
+    sequence = fields.Integer(default=10)
```
Sequencecomes generaly with the _order option. Dont' you think

ivantodorovich replied:
Oops! Forgot that one! Thanks!

legalsylvain commented on purchase_product_template_tags/__manifest__.py:
```diff
@@ -0,0 +1,16 @@
+# Copyright 2021 Camptocamp SA
+# @author Iván Todorovich <ivan.todorovich@gmail.com>
```
If the name is product_tempkate_tags_purchase, le module will be near in the folder.

ivantodorovich replied:
Yes but that's kind of breaking the guidelines

https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#id6
> When combining an Odoo module with another from OCA, Odoo's name goes before. I.e., if you want to combine Odoo's crm with OCA's partner_firstname, the name should be crm_partner_firstname.

legalsylvain replied:
Ow i didn't know that rule. Dont like it ! But ok !

simahawk replied:
I don't like it neither :rofl: 

but let's stick to it :wink:

ivantodorovich replied:
Yeah, I'm with you too :sweat_smile: 
Maybe we can propose a change in the guidelines. Like, all of the above and:

> [...] unless it's a very small glue module, in which case it's better to keep them closer together when sorted alphabetically

:smile:

legalsylvain reviewed (commented):
Minor remarks / partial review.

simahawk reviewed (changes requested):
nice work, thanks!

Please split changes in different commits, like:

1. 1 commit per new module eg: `Add purchase_product_template_tags`
2. 1 commit for the fix of the company rule
3. 1 commit for all the other UI improvements on the base module

ivantodorovich committed:
[IMP] product_template_tags: Add sequence

Allow to order tags.

ivantodorovich committed:
[FIX] product_template_tags: Multi-Company ir.rules

ivantodorovich committed:
[IMP] product_template_tags: Usability improvements

* Improve tags form view
* Improve tags search view: company_id should only be visible in case of multi-company
* Improve tree view: Replace products_count with color
* Improve product kanban's views
* Add a kanban view for product.product
* Add missing xml header in xml files
* Rename field from 'Product Tags' to 'Tags'

ivantodorovich committed:
[IMP] product_template_tags: Remove menuitem in general settings

This should be handled in glue modules for each
Sale, Purchase and Inventory apps.

ivantodorovich committed:
[ADD] sale_product_template_tags

ivantodorovich committed:
[ADD] purchase_product_template_tags

ivantodorovich committed:
[ADD] stock_product_template_tags

simahawk reviewed (approved):
@legalsylvain good for you?

simahawk commented:
/ocabot merge minor
